transit_odp/timetables/dataframes.py

```diff
@@ -256,32 +258,121 @@ def vehicle_journeys_to_dataframe(
+            ["OperatingProfile"]
         )
-        if serviced_organisation_day_type:
-            days_of_operation = serviced_organisation_day_type.get_element_or_none(
-                ["DaysOfOperation"]
-            )
-            days_of_non_operation = serviced_organisation_day_type.get_element_or_none(
-                ["DaysOfNonOperation"]
+        if operating_profile_vehicle_journey:
+            serviced_organisation_day_type = (
+                operating_profile_vehicle_journey.get_element_or_none(
+                    ["ServicedOrganisationDayType"]
+                )
             )
-            if days_of_operation:
-                operational = True
-                working_days = days_of_operation.get_element("WorkingDays")
-            elif days_of_non_operation:
-                operational = False
-                working_days = days_of_non_operation.get_element("WorkingDays")
-            serviced_org_ref = working_days.get_element("ServicedOrganisationRef").text
-
-            serviced_orgs_refs_df.append(
-                {"serviced_org_ref": serviced_org_ref, "operational": operational}
+            regular_day_type = operating_profile_vehicle_journey.get_element_or_none(
+                ["RegularDayType"]
             )
+            if regular_day_type:
+                days_of_week_elements = regular_day_type.get_elements_or_none(
+                    ["DaysOfWeek"]
+                )
+                if days_of_week_elements:
+                    days_of_week_element = days_of_week_elements[0]
+                    days_of_week = [
+                        day.localname for day in days_of_week_element.children
+                    ]
+            if serviced_organisation_day_type:
+                days_of_operation = serviced_organisation_day_type.get_element_or_none(
+                    ["DaysOfOperation"]
+                )
+                days_of_non_operation = (
+                    serviced_organisation_day_type.get_element_or_none(
+                        ["DaysOfNonOperation"]
+                    )
+                )
+                if days_of_operation:
+                    operational = True
+                    working_days = days_of_operation.get_element("WorkingDays")
+                elif days_of_non_operation:
+                    operational = False
+                    working_days = days_of_non_operation.get_element("WorkingDays")
+                serviced_org_ref = working_days.get_element(
+                    "ServicedOrganisationRef"
+                ).text
+        else:
+            for service in services:
+                service_code = service.get_element(["ServiceCode"]).text
+                if service_code == service_ref:
+                    operating_profile_services = service.get_element_or_none(
+                        ["OperatingProfile"]
+                    )
+                    if operating_profile_services:
+                        serviced_organisation_day_type = (
```

Comment:
The block of code starting from this line to end of if condition can be refactored into a separate method that can be reused for vehcile journey flow too